[[ Community Docs ]] URL Keyword enhancements

asayd opened the pull request:
- Added references;
- Added links;
- Fixed typos.

asayd committed:
[[ Community Docs ]] URL Keyword enhancements

- Added references;
- Added links;
- Fixed typos.

livecodeali commented on docs/dictionary/keyword/URL.lcdoc:
```diff
@@ -43,27 +44,45 @@ on textRead
+To ensure that these <URL(glossary)> types work in a desktop <standalone application>, you must include this custom <library> when you 
+create your standalone. In the Inclusions section of the Standalone Application Settings window, make sure Internet Library is 
+selected in the list of script libraries. The iOS and Android engines do not support the libUrl <library> but allow you to use 
+<URL(keyword)>. When specifying <URL(glossary)|URLs> for iOS and Android, you must use the appropriate form that confirms to 
```
There's a `confirms` instead of `conforms` here (again preexisting)

livecodeali commented on docs/dictionary/keyword/URL.lcdoc:
```diff
@@ -43,27 +44,45 @@ on textRead
+[RFC 1630](https://tools.ietf.org/html/rfc1630). 
+Ensure that you <urlEncode> any username and password fields appropriately for <FTP> urls.
+
+>*Important:*  The space character is not valid in <url(glossary)|URLs>, however the libURL <library> (Desktop platforms) replaces 
```
I guess this `<url(glossary)` needs to be `<URL(glossary)` due to case sensitivity in the javascript

livecodeali commented on docs/dictionary/keyword/URL.lcdoc:
```diff
@@ -43,27 +44,45 @@ on textRead
-From LiveCode 7.0.0, URL keyword has been upgraded to understand Unicode files when using URL("file:"). If the file located at the path has a Byte Order Mark at its beginning, then URL will decode the file according to that Byte Order Mark (UTF-8, UTF-16BE, UTF-16LE, UTF-32BE and UTF-32LE are compatible). If no Byte Order Mark is found, then the file will be decoded as if it were using Native enc ...
-
-References: HTMLText (property), URL (keyword), file (keyword), libURLSetAuthCallback (command), launch url (command), libURLSetExpect100 (command), load (command), container (glossary), keyword (glossary), resource (glossary)
+Use the <URL(keyword)> <keyword> to access the contents of a <file|local file> or a <file> accessible on the Web.
```
There's actually a `<local file>` glossary entry - this should probably link to that instead, and I guess the local file referenced in the summary too

livecodeali commented:
I think some (maybe most?) of the occurrences of `<file>` here should be linking to the file glossary entry rather than the keyword. What do you think?

asayd committed:
Update URL.lcdoc

@livecodeali Corrected the things you pointed out. Concur with your suggestion re file(glossary) vs file(keyword).  Sorted References alphabetically for better sorting in LC dictionary. Tell me if this is a bad idea.

livecodeali commented:
@asayd That looks great. Unfortunately you seem to have replied to my comment within the commit message of that update - we'll need to find a way to change that so that it describes the actual file changes. I think there are two options:

1) Submit a new pull request with these changes (you can copy the modified lcdoc text from this PR from here https://github.com/asayd/livecode/blob/patch-4/docs/dictionary/keyword/URL.lcdoc)

2) Use `git commit --amend` on the command line to modify the last commit message

For option 2) you'd have to clone your fork of the repo

```
git clone --recursive https://github.com/asayd/livecode.git
```

`cd` into the newly created livecode directory and checkout this branch

```
git checkout patch-4
```

amend the commit message

```
git commit --amend
```

(this will open up a text editor (probably vim) and allow you to edit the commit message)
and then push

```
git push --force
```

(you have to force push because you're overwriting the existing commit)

Let me know if you need a hand with either of these!

asayd commented:
@livecodeali - For me option 1 seems easiest. I'll close this PR and resubmit.
Working on Github marketing motto: "Now with more ways to screw up!" :-)
